astropy/coordinates/sky_coordinate_parsers.py

```diff
@@ -588,7 +588,19 @@ def _get_representation_attrs(frame, units, kwargs):
     for frame_attr_name, repr_attr_class, unit in zip(frame_attr_names, repr_attr_classes, units):
         value = kwargs.pop(frame_attr_name, None)
         if value is not None:
-            valid_kwargs[frame_attr_name] = repr_attr_class(value, unit=unit)
+            try:
+                valid_kwargs[frame_attr_name] = repr_attr_class(value, unit=unit)
+            except u.UnitConversionError as err:
+                error_message = "Unit '{unit}' ({type}) could not be applied to '{name}'. ".format(
```

Comment:
Just use an `f-string`.

Reply:
Right, I am so used to write Python 3.4 code I suppose :sweat_smile: will do